pkg/es/config/config.go

```diff
@@ -188,10 +202,65 @@ func (c *Configuration) GetTagDotReplacement() string {
 }
 
 // GetConfigs wraps the configs to feed to the ElasticSearch client init
-func (c *Configuration) GetConfigs() []elastic.ClientOptionFunc {
+func (c *Configuration) GetConfigs(logger *zap.Logger) []elastic.ClientOptionFunc {
 	options := make([]elastic.ClientOptionFunc, 3)
+	if c.TLS.Enabled {
+		tlsConfig, err := c.CreateTLSConfig(logger)
+		if err != nil {
+			return nil
+		}
+		httpClient := &http.Client{
+			Timeout: c.Timeout,
+			Transport: &http.Transport{
+				TLSClientConfig: tlsConfig,
+			},
+		}
+		options[0] = elastic.SetHttpClient(httpClient)
+		options[1] = elastic.SetURL(c.Servers...)
+		options[2] = elastic.SetSniff(c.Sniffer)
+		return options
+	}
 	options[0] = elastic.SetURL(c.Servers...)
 	options[1] = elastic.SetBasicAuth(c.Username, c.Password)
 	options[2] = elastic.SetSniff(c.Sniffer)
 	return options
 }
+
+// CreateTLSConfig creates TLS Configuration to connect with ES Cluster.
+func (c *Configuration) CreateTLSConfig(logger *zap.Logger) (*tls.Config, error) {
+	rootCerts, err := c.LoadCertificatesFrom()
+	if err != nil {
+		logger.Fatal("Couldn't load root certificate", zap.Error(err))
+	}
+	if len(c.TLS.CertPath) > 0 && len(c.TLS.KeyPath) > 0 {
+		clientPrivateKey, err := c.LoadPrivateKeyFrom()
+		if err != nil {
+			logger.Fatal("Couldn't setup client authentication", zap.Error(err))
+		}
+		return &tls.Config{
+			RootCAs:      rootCerts,
+			Certificates: []tls.Certificate{*clientPrivateKey},
+		}, err
+	}
+	return nil, err
+}
+
+// LoadCertificatesFrom is used to load root certification
+func (c *Configuration) LoadCertificatesFrom() (*x509.CertPool, error) {
+	caCert, err := ioutil.ReadFile(c.TLS.CaPath)
+	if err != nil {
+		return nil, err
+	}
+	certificates := x509.NewCertPool()
+	certificates.AppendCertsFromPEM(caCert)
+	return certificates, nil
+}
+
+// LoadPrivateKeyFrom is used to load the private certificate and key for TLS
+func (c *Configuration) LoadPrivateKeyFrom() (*tls.Certificate, error) {
```

Comment:
Perhaps remove the "From" from the function name?

Reply:
Thanks for your comments. I will update